checker/aws/s3/s3.go

```diff
@@ -0,0 +1,125 @@
+// Copyright 2019-2025 The Wait4X Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package s3
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+	"strings"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/service/s3"
+	"wait4x.dev/v3/checker"
+)
+
+// Checker is a struct that implements the checker.Checker interface
+type Checker struct {
+	bucketName string
+	region     string
+	client     *s3.Client
+}
+
+// New creates a new S3 bucket checker.
+func New(bucketPath string, opts ...Option) *Checker {
+	bucketName := parseBucketName(bucketPath)
+
+	checker := &Checker{
+		bucketName: bucketName,
+	}
+
+	for _, opt := range opts {
+		opt(checker)
+	}
+
+	return checker
+}
+
+// parseBucketName extracts the bucket name from a given S3 bucket path.
+func parseBucketName(bucketPath string) string {
+	if strings.HasPrefix(bucketPath, "s3://") {
```

Comment:
Is `s3://` a well-known `DSN`? If not, only support the bucket name or bucket `ARN`.